coreplugins/updater.py

```diff
@@ -0,0 +1,442 @@
+        :return: True if there was a successful update, False if not
+        """
+
+        try:
+            f = open(TAGFILE)
+        except FileNotFoundError:
+            logging.getLogger(__name__).debug("I was not !update'd.")
+            return False
+        except IsADirectoryError:
+            logging.getLogger(__name__).error(
+                "{} is a directory, I expected a file or nothing. Please clean this up.".format(TAGFILE))
+            return False
+
+        lines = f.readlines()
+        if len(lines) > 1:
+            logging.getLogger(__name__).error("{} has more than 1 line. This should not happen.".format(TAGFILE))
+            return False
+        if len(lines) == 0 or lines[0].strip() == "":
+            logging.getLogger(__name__).error("{} is empty. This should not happen.".format(TAGFILE))
+            return False
+
+        if lines[0].strip() == ERRORCODE:
+            await utils.write_debug_channel(self.bot, "The update failed. I have no idea why. Sorry, master!")
+            os.remove(TAGFILE)
+            return False
+        else:
+            logging.getLogger(__name__).debug("I was !update'd! Yay!.")
+            await utils.write_debug_channel(
+                self.bot, "I updated successfully! One step closer towards world dominance!")
+            os.remove(TAGFILE)
+            return True
+
+    @commands.command(name="news", help="Presents the latest update notes.")
+    async def news(self, ctx):
+        await self.update_news(ctx.message.channel)
+
+    @commands.command(name="version", help="Returns the running bot version.")
+    async def version(self, ctx):
+        """Returns the version"""
+        await ctx.send(lang["version"].format(Config().VERSION))
+
+    @commands.command(name="replace", help="Confirms an !update command.")
+    async def confirm(self, ctx):
+        # Check if there is an update request running
+        if self.waiting_for_confirm is None:
+            return
+
+        # Check if chan and user is the same
+        chancond = self.waiting_for_confirm.channel == ctx.message.channel
+        usercond = self.waiting_for_confirm.author == ctx.message.author
+        if not (chancond and usercond):
+            return
+
+        await utils.log_to_admin_channel(self.to_log)
+        self.to_log = None
+        self.state = State.CONFIRMED
+
+    @commands.command(name="update", help="Updates the bot if an update is available")
+    async def update(self, ctx, *args):
+        # Argument parsing
+        if "check" in args and len(args) == 1:
+            release = self.check_release()
+            if release is None:
+                await ctx.message.channel.send(lang["no_new_version"])
+            else:
+                await ctx.message.channel.send(lang["new_version"].format(release))
+            return
+        elif len(args) != 0:
+            return
+
+        if not permChecks.check_full_access(ctx.author):
+            return
```

Comment:
Better `raise commands.MissingAnyRole([Config().ADMIN_ROLE_ID, Config().BOTMASTER_ROLE_ID])`

Reply:
Will be fixed later